src/code/z_sram_NES.c

```diff
@@ -1,67 +1,1643 @@
+ */
+void Sram_GenerateRandomSaveFields(void) {
+    s32 randBombers;
+    s16 sp2A;
+    s16 pad;
+    s16 i;
+    s16 j;
+    s32 k;
+    s16 randSpiderHouse;
+
+    func_80143A54();
+
+    gSaveContext.save.lotteryCodes[0][0] = Rand_S16Offset(0, 10);
+    gSaveContext.save.lotteryCodes[0][1] = Rand_S16Offset(0, 10);
+    gSaveContext.save.lotteryCodes[0][2] = Rand_S16Offset(0, 10);
+    gSaveContext.save.lotteryCodes[1][0] = Rand_S16Offset(0, 10);
+    gSaveContext.save.lotteryCodes[1][1] = Rand_S16Offset(0, 10);
+    gSaveContext.save.lotteryCodes[1][2] = Rand_S16Offset(0, 10);
+    gSaveContext.save.lotteryCodes[2][0] = Rand_S16Offset(0, 10);
+    gSaveContext.save.lotteryCodes[2][1] = Rand_S16Offset(0, 10);
+    gSaveContext.save.lotteryCodes[2][2] = Rand_S16Offset(0, 10);
+
+    // Needed to match...
+    for (i = 0; i < 3; i++) {
+        for (j = 0; j < 3; j++) {}
+    }
+
+    i = 0;
+    sp2A = Rand_S16Offset(0, 16) & 3;
+    k = 6;
+    while (i != k) {
+        randSpiderHouse = Rand_S16Offset(0, 16) & 3;
+        if (sp2A != randSpiderHouse) {
+            gSaveContext.save.spiderHouseMaskOrder[i] = randSpiderHouse;
+            i++;
+            sp2A = randSpiderHouse;
+        }
+    }
+
+    do {
+        randBombers = Rand_S16Offset(0, 6);
+    } while (randBombers <= 0 || randBombers >= 6);
+
+    gSaveContext.save.bomberCode[0] = randBombers;
+
+    i = 1;
+    while (i != 5) {
+        k = false;
+
+        do {
+            randBombers = Rand_S16Offset(0, 6);
+        } while (randBombers <= 0 || randBombers >= 6);
+
+        sp2A = 0;
+        do {
+            if (randBombers == gSaveContext.save.bomberCode[sp2A]) {
+                k = true;
+            }
+            sp2A++;
+        } while (sp2A < i);
+
+        if (k == false) {
+            gSaveContext.save.bomberCode[i] = randBombers;
+            i++;
+        }
+    }
+}
+
+SavePlayerData sSaveDefaultPlayerData = {
+    { '\0', '\0', '\0', '\0', '\0', '\0' },             // newf
+    0,                                                  // deaths
+    { 0x3E, 0x3E, 0x3E, 0x3E, 0x3E, 0x3E, 0x3E, 0x3E }, // playerName "        "
+    0x0030,                                             // healthCapacity
+    0x0030,                                             // health
+    0x00,                                               // magicLevel
+    0x30,                                               // magic
+    0x0000,                                             // rupees
+    0x0000,                                             // swordHealth
+    0x0000,                                             // tatlTimer
+    0x00,                                               // magicAcquired
+    0x00,                                               // doubleMagic
+    0x00,                                               // doubleDefense
+    0x00,                                               // unk_1F
+    0xFF,                                               // unk_20
```

Comment:
```suggestion
    0,                                               // magicLevel
    0x30,                                               // magic
    0,                                             // rupees
    0,                                             // swordHealth
    0,                                             // tatlTimer
    0,                                               // magicAcquired
    0,                                               // doubleMagic
    0,                                               // doubleDefense
    0,                                               // unk_1F
    0xFF,                                               // unk_20
```
A lot of these can be shortened